lib/test.js

```diff
@@ -812,18 +880,21 @@ function notDeepLooseEqual(a, b, msg, extra) {
 Test.prototype.notDeepLooseEqual
 = notDeepLooseEqual;
 
+/** @type {import('./test').prototype.throws} */
 Test.prototype['throws'] = function (fn, expected, msg, extra) {
 	if (typeof expected === 'string') {
 		msg = expected;
 		expected = undefined;
 	}
 
+	/** @type {undefined | { error: unknown | Error }} */
 	var caught;
 
 	try {
 		fn();
 	} catch (err) {
 		caught = { error: err };
+		// @ts-expect-error TS doesn't understand that `Object(err) === err` narrows `err` to `object`
```

Comment:
What in tarnation?! TIL, thank you. 🤯 

But, what if it's an instance of `Array` or any other subclass of `Object`? (see other comment - is that possible?)

Reply:
`Object(x) === x` should narrow the same way as `x && (typeof x === 'object' || typeof x === 'function')`